proposals/nnnn-rpc-design-refactoring.md

```diff
@@ -0,0 +1,147 @@
+# RPC design refactoring
+
+* Proposal: [SDL-NNNN](nnnn-rpc-design-refactoring.md)
+* Author: [Alexander Kutsan](https://github.com/LuxoftAKutsan)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+
+Rework design of application manager layer to:
+ - Reduce coupling between components
+ - Reduce responsibility of ApplicationManager class
+ - Provide plugin architecture for easy adding additional functionality and RPC groups   
+
+## Motivation
+
+SDL core project is actively implements new features and new functionality. 
+To keep project on track and reduce rick of regression during implementation
+some refactoring of application manager layer should be done. 
+
+### Reduce coupling between components and use dependency injection practice
+
+Now Application manager is service locator and need to passed in each component
+even if component does not need application manager, but need some other classes that owned by application manager.
+During construction components should explicitly receive all dependencies. 
+It will prevent unwanted access and improve encapsulation. 
+
+
+### Reduce responsibility of ApplicationManager class 
+Not application manager class have to many responsibilities:
```

Comment:
@LuxoftAKutsan `Not` or `Now`?

Reply:
@LuxoftAKutsan  Not or Now? 
please, clarify and edit if needed